pkg/alertmanager/amcfg.go

```diff
@@ -166,12 +167,12 @@ func (c alertmanagerConfig) String() string {
 type configGenerator struct {
 	logger  log.Logger
 	kclient kubernetes.Interface
-	store   *assetStore
+	store   *assets.AssetStore
 }
 
-func newConfigGenerator(logger log.Logger, kclient kubernetes.Interface, store *assetStore) *configGenerator {
+func newConfigGenerator(logger log.Logger, kclient kubernetes.Interface, store *assets.AssetStore) *configGenerator {
```

Comment:
We can simplify and require a non `nil` store here. `kclient` isn't used anywhere I think so it can be removed from the `configGenerator` struct.

```suggestion
func newConfigGenerator(logger log.Logger, store *assets.AssetStore) *configGenerator {
```